kubespawner/spawner.py

```diff
@@ -1127,3 +1216,34 @@ def get_args(self):
                 args[i] = '--hub-api-url="%s"' % (self.accessible_hub_api_url)
                 break
         return args
+
+    def _options_form_default(self):
+        if not self.profile_list:
+            return
+        temp_keys = [
+            {
+                'display': p.get('display_name', self.UNDEFINED_DISPLAY_NAME),
```

Comment:
If display name is undefined, we should throw an exception and abort early - since that's something the admin has to fix. I think making that easier for the admin to see rather than wait for the user to see is better.